<a href="https://colab.research.google.com/github/ArsyadAptaa/Tugas-KKA-Praktikum-EDA/blob/main/Praktik_Modul_EDA_Individu.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np

harga = np.array([5000, 7000, 3000, 12000, 4500])

print('Rata-rata harga:', harga.mean())
print('Harga tertinggi:', harga.max())
print('Harga setelah diskon 10%:', harga * 0.9)

Rata-rata harga: 6300.0
Harga tertinggi: 12000
Harga setelah diskon 10%: [ 4500.  6300.  2700. 10800.  4050.]


### Tugas Analisis 1

**Pertanyaan:**
Bandingkan hasil `harga * 0.9` di atas dengan jika harga berupa list Python biasa (bukan array). Apa yang terjadi bila kamu mencoba `[5000,7000,3000] * 0.9`? Jelaskan mengapa berbeda.

**Jawaban:**
Pada NumPy array, `harga * 0.9` dapat menghitung harga setelah diskon 10% untuk semua data sekaligus. Hasilnya adalah `[4500., 6300., 2700., 10800., 4050.]`. Sedangkan pada list Python biasa, `[5000,7000,3000] * 0.9` menghasilkan error karena list Python tidak dapat dikalikan langsung dengan bilangan desimal. Perbedaannya karena NumPy array mendukung operasi matematika secara langsung pada setiap elemen.


In [ ]:
import pandas as pd

data_kantin = {
    'menu': ['Nasi Goreng', 'Es Teh', 'Mie Ayam', 'Es Teh', None],
    'harga': [12000, 4000, 10000, 4000, 8000],
    'terjual': [23, 40, None, 35, 18]
}

df = pd.DataFrame(data_kantin)

print(df)

          menu  harga  terjual
0  Nasi Goreng  12000     23.0
1       Es Teh   4000     40.0
2     Mie Ayam  10000      NaN
3       Es Teh   4000     35.0
4         None   8000     18.0


### Tugas Analisis 2

**Pertanyaan:**
Amati tabel yang dihasilkan. Kolom mana yang terlihat memiliki data kosong (None)? Menurutmu apa risikonya jika data ini langsung dianalisis tanpa dibersihkan?

**Jawaban:**
Berdasarkan data pada Latihan 2, terdapat data kosong pada kolom `menu` dan `terjual`. Kolom `menu` memiliki satu data `None`, sedangkan kolom `terjual` juga memiliki satu data kosong. Jika data langsung dianalisis tanpa dibersihkan, hasil analisis dapat menjadi kurang tepat karena terdapat informasi yang belum lengkap.


In [ ]:
df = pd.read_csv('data_kantin.csv')

print(df.head())
print(df.info())
print(df.describe())
print(df.shape)

      tanggal         menu kategori  harga  terjual
0  2026-08-20   Teh Hangat  Minuman   3000     11.0
1  2026-08-31  Nasi Goreng  Makanan  12000     29.0
2  2026-08-21  Nasi Goreng  Makanan  12000     28.0
3  2026-08-28   Teh Hangat  Minuman   3000     12.0
4  2026-08-21     Es Jeruk  Minuman   5000     19.0
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tanggal   110 non-null    object 
 1   menu      107 non-null    object 
 2   kategori  110 non-null    object 
 3   harga     110 non-null    int64  
 4   terjual   104 non-null    float64
dtypes: float64(1), int64(1), object(3)
memory usage: 4.4+ KB
None
              harga     terjual
count    110.000000  104.000000
mean    6336.363636   22.923077
std     3520.001232   10.831178
min     2000.000000    8.000000
25%     3000.000000   13.000000
50%     5000.000000   20.000000
75%    10000.000000 

### Tugas Analisis 4

**Pertanyaan:**
Dari hasil df.info(), kolom mana yang jumlah non-null-nya lebih sedikit dari jumlah baris total? Apa artinya?

**Jawaban:**
Dari hasil df.info(), kolom menu dan terjual memiliki jumlah data non-null yang lebih sedikit dari jumlah baris total, yaitu 110 baris. Kolom menu memiliki 107 data non-null sehingga terdapat 3 data kosong, sedangkan kolom terjual memiliki 104 data non-null sehingga terdapat 6 data kosong. Artinya, kedua kolom tersebut memiliki missing value atau data kosong.

In [ ]:
print(df.isnull().sum())

df['terjual'] = df['terjual'].fillna(0)

df = df.dropna(subset=['menu'])

tanggal     0
menu        3
kategori    0
harga       0
terjual     6
dtype: int64


### Tugas Analisis 4

**Pertanyaan:**
Mengapa pada contoh di atas kolom terjual diisi (fillna) sedangkan baris dengan menu kosong justru dihapus (dropna)? Diskusikan alasan logisnya.

**Jawaban:**
Pada hasil pemeriksaan data, terdapat 6 data kosong pada kolom terjual dan 3 data kosong pada kolom menu. Kolom terjual diisi dengan 0 menggunakan fillna karena data kosong pada jumlah penjualan dapat dianggap sebagai tidak ada jumlah penjualan yang tercatat. Sedangkan baris dengan menu kosong dihapus menggunakan dropna karena nama menu merupakan informasi penting untuk melakukan analisis berdasarkan menu. Jika nama menu kosong, data tersebut sulit digunakan untuk mengetahui produk yang dianalisis.

In [ ]:
print(df.duplicated().sum())

df = df.drop_duplicates()

df['harga'] = df['harga'].astype(int)

print(df.dtypes)

5
tanggal      object
menu         object
kategori     object
harga         int64
terjual     float64
dtype: object


### Tugas Analisis 5

**Pertanyaan:**
Setelah drop_duplicates(), berapa jumlah baris dataset sekarang dibanding sebelumnya? Mengapa penting memastikan tipe data (dtypes) sudah benar sebelum data dianalisis lebih lanjut?

**Jawaban:**
Sebelum dan setelah drop_duplicates(), jumlah baris dataset tetap 5 baris karena tidak terdapat data duplikat. Memastikan tipe data (dtypes) sudah benar penting agar setiap kolom dapat digunakan sesuai fungsinya dalam proses analisis. Contohnya, kolom harga bertipe integer sehingga dapat digunakan dalam perhitungan.

In [ ]:
laris = df[df['terjual'] > 20]

urut = df.sort_values(by='terjual', ascending=False)

df['total_pendapatan'] = df['harga'] * df['terjual']

ringkasan = df.groupby('menu')['total_pendapatan'].sum()

print(ringkasan)

menu
Es Jeruk       1460000.0
Es Teh         2144000.0
Kerupuk         972000.0
Mie Ayam       2410000.0
Nasi Goreng    3924000.0
Roti Bakar     1288000.0
Teh Hangat      459000.0
Name: total_pendapatan, dtype: float64


### Tugas Analisis 6

**Pertanyaan:**
Dari hasil groupby di atas, menu apa yang menghasilkan total_pendapatan tertinggi? Bagaimana informasi ini bisa membantu pengambilan keputusan di kantin sekolah?

**Jawaban:**
Berdasarkan hasil groupby, menu yang menghasilkan total_pendapatan tertinggi adalah Nasi Goreng, yaitu sebesar Rp3.924.000.

Informasi ini dapat membantu kantin sekolah mengetahui menu yang menghasilkan pendapatan paling besar sehingga dapat menjadi bahan pertimbangan dalam mengatur penjualan dan persediaan menu.